### DISCUSSION TITLE
HuggingFace ECAPA model - What data were used for training?

### DISCUSSION THREAD

**Discussion by fra1993** [speechbrain/speechbrain#1846] [General]
Hi, did you use an external RIR dataset to train this model as it happens by default when retraining with the standard ECAPA receipe in SpeechBrain? Thanks.

**Comment by anautsch** [comment]
Hi @fra1993 to which of the ECAPA pretrained models are you referring to?

Please take a look through our recipes:
https://github.com/speechbrain/speechbrain/tree/develop/recipes

They are sorted by databse, then by tasks in nested folders. Usually, we put on HuggingFace only pretrained models from our recipes.

For example, the VoxCeleb one refers to:
https://github.com/speechbrain/speechbrain/blob/c7238432bbde1b511d83099675f7a97e05e10cea/recipes/VoxCeleb/SpeakerRec/hparams/train_ecapa_tdnn.yaml#L91-L99

So there's a hint to `speechbrain.lobes.augment`:
https://github.com/speechbrain/speechbrain/blob/c7238432bbde1b511d83099675f7a97e05e10cea/speechbrain/lobes/augment.py#L29

Hope this helps to read SpeechBrain code better.

**Comment by fra1993** [comment]
Hi, thanks for the answer. I was referring to speechbrain/spkrec-ecapa-voxceleb: the pretrained model for speaker verificaton with speaker embedding dimension = 192. I believe that this model uses an external RIR database for augmentation but I want to be 100% sure.

**Reply by anautsch** [reply]
@fra1993 I don't know how to reply to this better: the link is posted above.

Did you try debugging your Python application?
e.g. https://www.jetbrains.com/help/pycharm/debugging-your-first-python-application.html

Above I demonstrated to you how to do it manually, and since that is not a working method for you, I'd encourage you to use the debugger. Again, please check the links above, for it might outline already what you are asking for. For putting a debug point—as already suggested indirectly, but now more directly—please take a look and debug this function within the recipe you are curious about:
https://github.com/speechbrain/speechbrain/blob/055d465b2a2bb727c26b030b83a791f2a14d8dfc/speechbrain/lobes/augment.py#L302

To identify which recipe you are curious about...
1. here are all our recipes (basis for pretrained models) -> https://github.com/speechbrain/speechbrain/tree/develop/recipes
2. VoxCeleb -> https://github.com/speechbrain/speechbrain/tree/develop/recipes/VoxCeleb
3. hparams to check emb dim == 192 -> https://github.com/speechbrain/speechbrain/tree/develop/recipes/VoxCeleb/SpeakerRec/hparams
4. this folder has a [train_ecapa_tdnn.yaml](https://github.com/speechbrain/speechbrain/blob/develop/recipes/VoxCeleb/SpeakerRec/hparams/train_ecapa_tdnn.yaml) (same file as linked above)

https://github.com/speechbrain/speechbrain/blob/055d465b2a2bb727c26b030b83a791f2a14d8dfc/recipes/VoxCeleb/SpeakerRec/hparams/train_ecapa_tdnn.yaml#L66-L78

I hope this how-to guide can also serve as a reference point for others with similar question of how to link SpeechBrain  pretrained models on HuggingFace with SpeechBrain recipes to figure out some detail in the training procedure.

**Reply by fra1993** [reply]
Hi, yes I checked all these things when training my network. I guess I just wanted to be sure you used the RIR augmentation for the pretrained model you loaded on huggingface which looked like the case from the yaml file in the recipes folder but I needed to be sure. Thanks

**Reply by anautsch** [reply]
I did not write the recipe, I did only perform a quick check ... other recipes use additional noise sources. Please check which files are created on your local disk when running this recipe. Dummy data is enough, e.g., as of our recipe tests
https://github.com/speechbrain/speechbrain/tree/develop/tests/recipes

All recipes of the VoxCeleb dataset can be checked using this command:
```bash
python -c 'from tests.utils.recipe_tests import run_recipe_tests; print("TEST FAILED!") if not(run_recipe_tests(filters_fields=["Dataset"], filters=[["VoxCeleb"]], do_checks=False, run_opts="--device=cuda")) else print("TEST PASSED")'
```
@mravanelli did a great job last summer in outlining the first steps for such testing.

It's a few weeks ago since I worked on that closely, but it should download all the noises too which are necessary for mixing data etc.—and if not, depending overrides can be found under `test_debug_flags` here:
https://github.com/speechbrain/speechbrain/blob/develop/tests/recipes/VoxCeleb.csv

I understand the desire to be 100% — been there.

If you like to use SpeechBrain more frequent, take a pause to ponder, how the building blocks from yaml & train recipes play together. You'll need some fundament to stand on. Such, one  can only build for oneself.

Look, if you know where you are stuck—like at which point which method is lacking—, just ask about it. It will help you and others.
